### ISSUE TITLE
decode_compressed_response=True changes the original response header

### ISSUE THREAD

**Issue by lmazuel** [kevin1024/vcrpy#264]
If you use `decode_compressed_response=True`, you actually add the `vcr.filter.decode_reponse` as response filter.

Inside this method, you could find this:

``` python
    headers = CaseInsensitiveDict(response['headers'])
    if is_compressed(headers):
        response = copy.deepcopy(response)
        encoding = headers['content-encoding'][0]
        headers['content-encoding'].remove(encoding)
...
        response['headers'] = dict(headers)
```

The objective is to do a deepcopy of response of avoid to change it. However, it's changing the `headers` anyway.

Because the "headers" values are list and `CaseInsensitiveDict` does not do a copy of the values. Since the "remove(encoding)" is applied on the "headers" dict AFTER the deepcopy but on the "headers" created BEFORE, you change the actual headers list:

``` python
In [1]: from vcr.util import CaseInsensitiveDict
In [2]: response_header = {"content-encoding":['gzip']}
In [3]: headers = CaseInsensitiveDict(response_header)
In [4]: import copy
In [5]: new_response_header = copy.deepcopy(response_header)
In [6]: headers['content-encoding'].remove('gzip')
In [7]: response_header
Out[7]: {'content-encoding': []}
```

Recreating the CaseInsentiveDict after the deepcopy solves the problem.

**Comment by nickdirienzo** [comment]
@lmazuel hey there. I ran into the same issue as well a few months ago. I patched a fix to resolve this problem in #259 that's included in 1.9.0. Which version of vcrpy is your code snippet from?

**Comment by neozenith** [comment]
A lot of changes have happened to VCRpy since this ticket was opened. As this ticket has become stale and we have a lot of old tickets that need to be groomed, I'm closing this for now to make it easier to see what is still relevant.

However if it is still needed, please feel free to re-open or create a new ticket.

Thanks! 🙏
